### PR TITLE
Improve `get_model_dependencies` docstring

### LINKED ISSUE #9809: [DOC-FIX] mlflow.pyfunc.get_model_dependencies() downloads all the artifacts
### Willingness to contribute

Yes. I can contribute a documentation fix independently.

### URL(s) with the issue

Hello,
Here is the current link of this method documentation : https://mlflow.org/docs/latest/python_api/mlflow.pyfunc.html
My issue is that the documentation of the method `mlflow.pyfunc.get_model_dependencies` doesn't specify that it downloads all the artifacts of a given model (not only the `requirements.txt` file), which causes me some troubles (`Connection broken` and `ChunkedEncodingError`) when a large model (about 500MiB) is stored in the artifacts. 

I don't know if it's you're first intention to return all the artifacts or if this method should only download and return the `requirement.txt` file. To this extent, I propose you to add a little description to indicate its behavior. Otherwise, we could imagine to add an input parameter to choose whether the user needs to download all the artifacts or only the requirements.

### Description of proposal (what needs changing)

mlflow.pyfunc.get_model_dependencies(model_uri, format='pip')[[source]](https://mlflow.org/docs/latest/_modules/mlflow/pyfunc.html#get_model_dependencies)

**Proposal ---------------**
Download all the artifacts from the model_uri, store them in a `temp/` folder and return the path of the requirements.txt file of that folder.
**-------------------------**

Parameters
model_uri – The uri of the model to get dependencies from.

format – The format of the returned dependency file. If the "pip" format is specified, the path to a pip requirements.txt file is returned. If the "conda" format is specified, the path to a "conda.yaml" file is returned . If the "pip" format is specified but the model was not saved with a requirements.txt file, the pip section of the model’s conda.yaml file is extracted instead, and any additional conda dependencies are ignored. Default value is "pip".

Returns
The local filesystem path to either a pip requirements.txt file (if format="pip") or a conda.yaml file (if format="conda") specifying the model’s dependencies.

### PR DESCRIPTION
**Pull request by harupy**
[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/9847?quickstart=1)

### Related Issues/PRs

Resolve #9809

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/pyfunc/__init__.py [hunk 1]
```diff
@@ -842,6 +842,14 @@ def get_conda_yaml_path():
 
 def get_model_dependencies(model_uri, format="pip"):  # pylint: disable=redefined-builtin
     """
+    Downloads the model dependencies and returns the path to requirements.txt or conda.yaml file.
+
+    .. warning::
+        This API downloads all the model artifacts to the local filesystem. This may take
```
Comment (WeichenXu123) [inline mlflow/pyfunc/__init__.py:848]:
```suggestion
        This API downloads all the model artifacts to the local filesystem. This may takes
```

**Review by WeichenXu123 — APPROVED** [review]
LGTM

File: mlflow/pyfunc/__init__.py (hunk 1 above)
Comment (harupy) [reply mlflow/pyfunc/__init__.py:848]: /ai Is the suggestion correct?

Comment (harupy) [reply mlflow/pyfunc/__init__.py:848]: @WeichenXu123 Thoughts?

Comment (harupy) [reply mlflow/pyfunc/__init__.py:848]: /ai What form of verb should you use after modal verbs?

Comment (harupy) [reply mlflow/pyfunc/__init__.py:848]: 👆 @WeichenXu123

Comment (harupy) [reply mlflow/pyfunc/__init__.py:848]: https://prowritingaid.com/grammar/1000140/What-form-of-verb-should-you-use-after-modal-verbs

**Pull request merged**